packages/user-operation-controller/src/helpers/Bundler.ts

```diff
@@ -33,16 +44,30 @@ export class Bundler {
     userOperation: UserOperation,
     entrypoint: string,
   ): Promise<BundlerEstimateUserOperationGasResponse> {
+    log('Estimating gas', { url: this.#url, userOperation, entrypoint });
+
     const response = await this.#query('eth_estimateUserOperationGas', [
       userOperation,
       entrypoint,
     ]);
 
-    log('Estimated gas', { url: this.#url, response });
+    log('Estimated gas', { response });
 
     return response as BundlerEstimateUserOperationGasResponse;
   }
 
+  /**
+   * Retrieve the receipt for a user operation.
+   * @param hash - The hash of the user operation.
+   * @returns The receipt for the user operation.
+   */
+  async getUserOperationReceipt(
+    hash?: string,
+  ): Promise<UserOperationReceipt | undefined> {
```

Comment:
The receipt is only available once the user operation has been processed, successfully or otherwise, and so until that point the bundler RPC method returns `null` or `undefined`.

Some bundlers also throw an error if the receipt isn't available though that feels less accurate to the specification.

Will update the comment in the subsequent PR.